types/test.tsx

```diff
@@ -0,0 +1,34 @@
+import parse from 'html-dom-parser';
+import * as React from 'react';
+
+// $ExpectType string | DetailedReactHTMLElement<{}, HTMLElement> | DetailedReactHTMLElement<{}, HTMLElement>[]
+parse('<div>text</div>');
+
+// `options.replace`
+
+// Return React.createElement from `replace`
+parse('<p id="replace">text</p>', {
+  replace: domNode => {
+    if (domNode.attribs && domNode.attribs.id === 'replace') {
+      return React.createElement('span', {}, 'replaced');
+    }
+  }
+});
+
+// Return ReactElement
+parse('<p><br id="remove"></p>', {
+  replace: ({ attribs }) =>
+    attribs && attribs.id === 'remove' && <React.Fragment />
+});
+
+// Return domhandler node
+parse('<a href="#">Heading</a>', {
+  replace: node => {
+    if (node.attribs && node.attribs.id === 'header') {
+      return {
```

Comment:
Is this return unnecessary?

Reply:
Have no idea. I took this from `dom-to-react.js`

Reply:
I ask because since this returns only when there's an element with id of `header`, this if-check will get skipped entirely

Reply:
This code isn't actually called. It serves only as type checking sample

Reply:
I wonder what is type of the returned object though. I haven't figured out it from the source code

Reply:
You're right in that the return type is `any`. I could _technically_ restrict it but I decided to keep it flexible (I'm checking the return value with `React.isValidElement()`